lahja/endpoint.py

```diff
@@ -129,6 +198,68 @@ def connect_no_wait(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> N
 
             self._process_item(item, config)
 
+    def _create_external_api(self, ipc_path: pathlib.Path) -> None:
+
+        receiver = EndpointConnector(self)
+
+        class ConnectorManager(BaseManager):
+            pass
+
+        ConnectorManager.register('get_connector', callable=lambda: receiver)  # type: ignore
+
+        manager = ConnectorManager(address=str(ipc_path))  # type: ignore
+        server = manager.get_server()   # type: ignore
+        threading.Thread(target=server.serve_forever, daemon=True).start()
+
+    def connect_to_endpoints_blocking(self, *endpoints: ConnectionConfig, timeout: int=30) -> None:
+        """
+        Connect to the given endpoints and block until the connection to every endpoint is
+        established. Raises a ``TimeoutError`` if connections do not become available within
+        ``timeout`` seconds (default 30 seconds).
+        """
+        self._throw_if_already_connected(*endpoints)
+        for endpoint in endpoints:
+            wait_for_path_blocking(endpoint.path, timeout)
+            self._connect_to_endpoint(endpoint)
+
+    async def connect_to_endpoints(self, *endpoints: ConnectionConfig) -> None:
+        """
+        Connect to the given endpoints and await until all connections are established.
+        """
+        self._throw_if_already_connected(*endpoints)
+        await asyncio.gather(
+            *(self._await_connect_to_endpoint(endpoint) for endpoint in endpoints),
+            loop=self._loop
+        )
+
+    def connect_to_endpoints_nowait(self, *endpoints: ConnectionConfig) -> None:
+        """
+        Connect to the given endpoints as soon as they become available but do not block.
+        """
+        self._throw_if_already_connected(*endpoints)
+        for endpoint in endpoints:
+            asyncio.ensure_future(self._await_connect_to_endpoint(endpoint))
```

Comment:
Same here as in `connect_to_endpoints`, there's a race-condition between checking whether the name already exists and then claiming the name.